crates/file_finder/src/file_finder_tests.rs

```diff
@@ -2133,18 +2133,28 @@ async fn test_repeat_toggle_action(cx: &mut gpui::TestAppContext) {
 
     cx.dispatch_action(ToggleFileFinder::default());
     let picker = active_file_picker(&workspace, cx);
+
+    picker.update_in(cx, |picker, window, cx| {
+        picker.update_matches(".txt".to_string(), window, cx)
```

Comment:
This test was broken on main already, but we did not notice it. When the query is empty, the file finder only displays recent items, meaning that on `main`, the `delegate.matches.len()` was always `0`. 
There was a bug in the picker code which actually let you change the selected index to an arbitrary number, even if the matches were empty